lib/matplotlib/axes/_axes.py

```diff
@@ -8443,20 +8456,26 @@ def violin(self, vpstats, positions=None, vert=True, widths=0.5,
         # Check whether we are rendering vertically or horizontally
         if vert:
             fill = self.fill_betweenx
-            perp_lines = functools.partial(self.hlines, colors=linecolor)
-            par_lines = functools.partial(self.vlines, colors=linecolor)
+            perp_lines = functools.partial(self.hlines, colors=linecolor,
+                                           capstyle='projecting')
```

Comment:
To keep backward compatibility and likely prevent the regeneration of the reference images, I would change the capstyle only for the half-sided violins.